subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -178,12 +179,17 @@ public EntitlementState getState() {
         }
 
         final SubscriptionBaseTransition pendingTransition = getPendingTransition();
-        if (pendingTransition != null) {
-            return EntitlementState.PENDING;
+        if (pendingTransition != null && pendingTransition.getEffectiveTransitionTime().isAfter(clock.getUTCNow())) {
+        		return EntitlementState.PENDING;
         }
+        else if(pendingTransition.getEffectiveTransitionTime().isBefore(clock.getUTCNow()) && isFutureCancelled()){
```

Comment:
⚠️  One way to reach this statement if if `pendingTransition = null` and then we would NPE

However, based on previous point, I don't think we could have non null `pendingTransition` whose date is in the past.


Stepping back: Do you need to even change the code for your test to pass -- and knowing that we kept the semantics we already have?

Reply:
@sbrossie, yes you are right, the code does not need to be changed for the test to pass. So, does that mean that [1528 ](https://github.com/killbill/killbill/issues/1528) is not really a bug and the system behaves as expected?

Reply:
Yep, according to our semantics, this is not a bug: The main reason for choosing this semantics (this is a choice we have), is that we want the ability to uncancel (if needed) and if the state is `canceled` prior we reach the effective date, i.e the create/cancel date, we can't do it (hence your other test that was failing).

Reply:
okay, that makes sense. I will revert my code changes. So, should we close #1528?

Reply:
Let's complete this PR, and then as we merge the PR, you can close the ticket.

Reply:
Reverted changes to `DefaultSubscriptionBase`